hardware_interface/CMakeLists.txt

```diff
@@ -1,169 +1,204 @@
+    pluginlib)
+  install(TARGETS test_components
+    DESTINATION lib
+  )
+  pluginlib_export_plugin_description_file(
+    hardware_interface test/test_components/test_components.xml)
+
+  add_library(test_hardware_components SHARED
+  test/test_hardware_components/test_single_joint_actuator.cpp
+  test/test_hardware_components/test_force_torque_sensor.cpp
+  test/test_hardware_components/test_two_joint_system.cpp
+  )
+  target_link_libraries(test_hardware_components hardware_interface)
+  ament_target_dependencies(test_hardware_components
+    pluginlib)
+  install(TARGETS test_hardware_components
+    DESTINATION lib
+  )
+  pluginlib_export_plugin_description_file(
+    hardware_interface test/test_hardware_components/test_hardware_components.xml
+  )
+
+  ament_add_gmock(test_resource_manager test/test_resource_manager.cpp)
+  target_link_libraries(test_resource_manager hardware_interface)
+  ament_target_dependencies(test_resource_manager ros2_control_test_assets)
+
+  ament_add_gmock(test_generic_system test/fake_components/test_generic_system.cpp)
+  target_include_directories(test_generic_system PRIVATE include)
+  target_link_libraries(test_generic_system hardware_interface)
+  ament_target_dependencies(test_generic_system
+    pluginlib
+    ros2_control_test_assets
+  )
+
+  # joint_limiter tests
+  ament_add_gmock(test_joint_limiter_interface test/joint_limits/test_joint_limiter_interface.cpp)
+  target_include_directories(test_joint_limiter_interface PUBLIC include)
+  ament_target_dependencies(test_joint_limiter_interface rcutils)
+
+  # joint_limit_interface tests
+  ament_add_gmock(joint_limits_interface_test test/joint_limits/joint_limits_interface_test.cpp)
+  target_include_directories(joint_limits_interface_test PUBLIC include)
+  ament_target_dependencies(joint_limits_interface_test rclcpp)
+
+  add_executable(joint_limits_rosparam_test test/joint_limits/joint_limits_rosparam_test.cpp)
+  target_include_directories(joint_limits_rosparam_test PUBLIC include ${GTEST_INCLUDE_DIRS})
+  target_link_libraries(joint_limits_rosparam_test ${GTEST_LIBRARIES})
+  ament_target_dependencies(joint_limits_rosparam_test rclcpp)
+  add_launch_test(test/joint_limits/joint_limits_rosparam.launch.py)
+  install(
+          TARGETS
+          joint_limits_rosparam_test
+          DESTINATION lib/${PROJECT_NAME}
+  )
+  install(
+          FILES
+          test/joint_limits/joint_limits_rosparam.yaml
+          DESTINATION share/${PROJECT_NAME}/test
+  )
+
+  ament_add_gmock(joint_limits_urdf_test test/joint_limits/joint_limits_urdf_test.cpp)
+  target_include_directories(joint_limits_urdf_test PUBLIC include)
+  ament_target_dependencies(joint_limits_urdf_test rclcpp urdf)
+endif()
+
+ament_export_include_directories(
+  include
+)
+ament_export_libraries(
+  fake_components
+  hardware_interface
+)
+ament_export_dependencies(
+  control_msgs
+  pluginlib
+  rcpputils
+  tinyxml2_vendor
+  TinyXML2
+  urdf

```

Comment:
why is the entire file flagged up as changed? changes between windows line ending vs unix?